src/libprojectM/Renderer/hlslparser/src/Engine.cpp

```diff
@@ -59,8 +64,31 @@ bool String_EqualNoCase(const char * a, const char * b) {
 #endif
 }
 
+static inline double iss_strtod(const char * in, char ** end) {
+    double df;
+    std::istringstream iss(in);
+    iss.imbue(std::locale("C"));
+    iss >> df;
+    if(iss.fail()) {
+        *end = (char*)in;
```

Comment:
consider using something other than (char *) -- static_cast<> is ideal, thgen dynamic_cast<> then, reinterpret_cast<> (equivalent to doing a C style cast.)